src/sentry/api/endpoints/organization_events_root_cause_analysis.py

```diff
@@ -1,10 +1,74 @@
+from datetime import datetime
+
 from rest_framework.response import Response
+from snuba_sdk import Column, Function
 
 from sentry import features
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.organization_events import OrganizationEventsEndpointBase
+from sentry.search.events.builder import QueryBuilder
+from sentry.search.events.types import QueryBuilderConfig
+from sentry.snuba.dataset import Dataset
 from sentry.snuba.metrics_performance import query as metrics_query
+from sentry.utils.snuba import raw_snql_query
+
+
+def query_spans(transaction, regression_breakpoint, params):
+    selected_columns = [
+        "count(span_id) as span_count",
+        "sumArray(spans_exclusive_time) as total_span_self_time",
+        "array_join(spans_op) as span_op",
+        "array_join(spans_group) as span_group",
+        # want a single event id to fetch from nodestore for the span description
+        "any(id) as sample_event_id",
+    ]
+
+    builder = QueryBuilder(
+        dataset=Dataset.Discover,
+        params=params,
+        selected_columns=selected_columns,
+        equations=[],
+        query=f"transaction:{transaction}",
+        orderby=["span_op", "span_group", "total_span_self_time"],
+        limit=10000,
+        config=QueryBuilderConfig(
+            auto_aggregations=True,
+            use_aggregate_conditions=True,
+            functions_acl=[
+                "array_join",
+                "sumArray",
+                "percentileArray",
+            ],
+        ),
+    )
+
+    builder.columns.append(
+        Function(
+            "if",
+            [
+                Function(
+                    "greaterOrEquals", [Column("timestamp"), regression_breakpoint.isoformat()]
+                ),
```

Comment:
is this getting only spans after the breakpoint?

Reply:
No, what this is doing is adding a column that splits the aggregates into before and after the breakpoint, so we can do the comparisons and keep it all in one query

Reply:
I could name the `period` alias to be something more descriptive, like `after_breakpoint`

Reply:
ohhh `1` and `0`

Reply:
very cool

Reply:
for clarity, you could change it to return some other string instead of `0` and `1` which can be ambiguous.

Reply:
Sounds good, I might make `period` have "before" and "after" as possible values. I feel like that's a good direction in clarity if no one opposes
cc: @npusarla